packages/router/src/__tests__/analyzeRoutes.test.tsx

```diff
@@ -645,7 +642,7 @@ describe('AnalyzeRoutes: with homePage and Children', () => {
           },
           // ...and the third private set's props
           {
-            id: 4,
+            id: '0.0.1',
```

Comment:
Also what are your thoughts on incrementing the nextSetId after 0 so the numbers are `0.1.2` instead of `0.0.1` was a little confused in the tests

Reply:
Yeah, I wasn't sure what to do about the numbering. For the functionality it doesn't really matter, as long as it's unique for each Set. 

Could have done `id: uuid()`. That would have been the easiest way to do it. But testing would have been much more difficult because you'd never know what id each set would have gotten.

Could also have done `id: previousSets.at(-1)?.id + '.' + nextSetId`. Testable and would have worked just fine. But the ids for the root level would have ended up being `undefined.1` etc. Ugly!

Could also have gotten rid of `nextSetId` and just used the iteration index like `id: previousSets.at(-1)?.id + '.' + index`. But then the first id would have been `undefined.0`, and the second could have been `undefined.5` depending on how many regular routes there were in between... Again, simpler code, but ugly and not as straight forward to understand what id each Set would end up getting.

I did make one slight change as a result of your comment though – I now start numbering at 1. I want to pretend I'm a 1337 h4x0r and naturally start counting at 0, but who am I kidding? It is more natural for me to just start counting at 1.

So the way it works now is each level starts at 1 and increases as you go down the route tree on that same level. As soon as you go a level deeper counting restarts again at 1. So `'1.1.2` would be a Set that's nested three levels deep and it's the first Set on the first and second levels and the second set at the third level.

Reply:
Yeah I think that works fine. It makes sense and yeah I totally get the 1337 haxor feeling and I understood what was going on but for some reason when looking at the tests it wasn't clear. I think the root at each level makes sense.